core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -285,4 +285,11 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext {
     }
   }
 
+  test("No exception when both num-executors and dynamic allocation set") {
```

Comment:
can this test assert something stronger, that we actually end up with 6 executors?